Client/Teztez/Core/Backwards/Modules/BackwardsConfiguration/Controller/BackwardsConfigurationViewController.swift

```diff
@@ -0,0 +1,165 @@
+
+    func setUserText(_ text: String)
+    func setUserFontSize(_ fontSize: CGFloat)
+    func setupBackwardsPause()
+}
+
+private enum Constants {
+    static let spacing: CGFloat = 32.0
+}
+
+final class BackwardsConfigurationViewController: ViewController, BackwardsConfigurationPresentable {
+    var onCloseButtonDidTap: Callback?
+    var onTextInputDidTap: Callback?
+    var onFontSizeChangeDidTap: Callback?
+    var onStartButtonDidTap: ((Configuration) -> Void)?
+    var onContinueButtonDidTap: Callback?
+
+    private let store: BackwardsConfigurationStore
+    private var cancellables = Set<AnyCancellable>()
+
+    private lazy var headerView = ActivityHeaderView()
+    private lazy var inputTextView = ActivityTextInputView()
+    private lazy var selectFontSizeView = ActivitySelectSizeView()
+
+    @IBOutlet private var stackView: UIStackView!
+    @IBOutlet private var startButton: PrimaryButton!
+    @IBOutlet private var restartButton: SecondaryButton!
+
+    init(store: BackwardsConfigurationStore) {
+        self.store = store
+        super.init(nibName: String(describing: Self.self), bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        return nil
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupObservers()
+        setupUI()
+    }
+
+    @IBAction func startButtonDidTap(_ sender: PrimaryButton) {
+        switch sender.tag {
+        case 0:
+            store.dispatch(action: .didStartDidTap)
+        case 1:
+            onContinueButtonDidTap?()
+        default:
+            break
+        }
+    }
+
+    @IBAction func restartButtonDidTap(_ sender: SecondaryButton) {
+        store.dispatch(action: .didStartDidTap)
+    }
+
+    func setUserText(_ text: String) {
+        store.dispatch(action: .didSetUserText(text: text))
+    }
+
+    func setUserFontSize(_ fontSize: CGFloat) {
+        store.dispatch(action: .didSelectFontSize(fontSize: fontSize))
+    }
+
+    func setupBackwardsPause() {
+        startButton.setTitle(R.string.backwardsConvertText.continueTitle(), for: .normal)
+        startButton.tag = 1
+        restartButton.isHidden = false
+    }
+
+    private func setupObservers() {
+        store.$state.sink { [weak self] state in
+            guard let self = self, let state = state else { return }
+            switch state {
+            case let .initial(blocks):
+                self.setupViews(from: blocks)
+            case let .updated(block: block):
+                self.updateBlock(block)
+            case let .configured(configuration: configuration):
+                self.onStartButtonDidTap?(configuration)
+            }
+        }.store(in: &cancellables)
+    }
+
+    private func setupUI() {
+        store.dispatch(action: .didLoadView)
+        stackView.spacing = Constants.spacing
+        setupNavigationBar()
+    }
+
+    private func setupNavigationBar() {
+        navigationController?.view.backgroundColor = .systemGray
```

Comment:
we dont need to add this line, because navigationBar.barTintColor = .systemGray do same job